src/globals.c

```diff
@@ -212,13 +213,37 @@ bool set_format_string(char *userinput)
 	return true;
 }
 
+bool set_state_dir(char *dir)
+{
+	if (dir == NULL || dir[0] == '\0') {
+		return false;
+	}
+
+	/* TODO: more validation of input? */
+	if (dir[0] != '/') {
```

Comment:
Not really sure what extra validation we need here...

Restricting the value to a full path is good, and because we create the directory in create_required_dirs(), we don't need an existence check.